### DISCUSSION TITLE
Log-in by link in email

### DISCUSSION THREAD

**Discussion by karolzlot** [fastapi-users/fastapi-users#670] [Q&A]
Hi. I want to have web app which uses links in emails. When use clicks such link, user is automatically logged in.

So I won't use passwords.

Can I achieve this behavior with this library or maybe I will be better off with something different?

**Comment by frankie567** [comment]
Hi there,

Sorry for the delayed response 🙂

I think you can achieve that, but that would require quite some work on your side:

- Reimplement a `/register` route that don't require a password. You can draw inspiration from the built-in one: https://github.com/frankie567/fastapi-users/blob/master/fastapi_users/router/register.py
- Implement an authentication route that takes your "magic link" in request param and generates a cookie. You can use the built-in [`CookieAuthentication` backend](https://github.com/frankie567/fastapi-users/blob/master/fastapi_users/authentication/cookie.py) and use its `get_login_response` method to do that. Then, your user will be authenticated for the session using a cookie.

**Reply by karolzlot** [reply]
Thank you for your answer 😃 

I see, I think I will try to use Local Storage instead (from React frontend).
And after user is authenticated, I will redirect to another page.
